cmd/main.go

```diff
@@ -332,21 +337,26 @@ func setupReconcilers(mgr ctrl.Manager, setupLog logr.Logger,
 		return err
 	}
 
-	inferenceGraphCrdAvailable, igCrdErr := utils.IsCrdAvailable(
-		mgr.GetConfig(),
-		v1alpha1.SchemeGroupVersion.String(),
-		"InferenceGraph")
-	if igCrdErr != nil {
-		setupLog.Error(igCrdErr, "unable to check if InferenceGraph CRD is available", "controller", "InferenceGraph")
-		return igCrdErr
-	} else if inferenceGraphCrdAvailable {
-		if err := setupInferenceGraphReconciler(mgr); err != nil {
-			setupLog.Error(err, "unable to create controller", "controller", "InferenceGraph")
-			return err
+	if kserveState == "managed" {
+		inferenceGraphCrdAvailable, igCrdErr := utils.IsCrdAvailable(
+			mgr.GetConfig(),
+			v1alpha1.SchemeGroupVersion.String(),
+			"InferenceGraph")
+		if igCrdErr != nil {
+			setupLog.Error(igCrdErr, "unable to check if InferenceGraph CRD is available", "controller", "InferenceGraph")
+			return igCrdErr
+		} else if inferenceGraphCrdAvailable {
+			if err := setupInferenceGraphReconciler(mgr); err != nil {
+				setupLog.Error(err, "unable to create controller", "controller", "InferenceGraph")
+				return err
+			}
+		} else {
+			setupLog.Info("crds unavailable, skipping controller", "controller", "InferenceGraph")
```

Comment:
IMO, the CRD check is no longer needed and we should always start the InferenceGraph controller when KServe is enabled. It is expected that the CRD is present. If it isn't present, there is a setup issue.